### PR TITLE
Add doc

### LINKED ISSUE #102: Add documentation page / Better showcase for API/data
It would be nice to have a nice documentation page about the API on the front-end instead, for a better showcase of the API and open data used on this front.

Some interesting examples :
https://adresse.data.gouv.fr/
https://geo.api.gouv.fr/

### PR DESCRIPTION
**Pull request by Samuelfaure**
fix #102 

This is a first version of the documentation page of the API for the website.

@thimy your request would be particularly important on this PR, I point your attention to the file ApiDoc.vue, which contains a lot of HTML and CSS code.

My integration skill aren't yet very good so I might have done some mistakes there and there.
Same for design skills, your feedback is very appreciated.

Screenshot :
firstdraftdoc

### CODE REVIEW

File: src/components/ApiDoc.vue [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+          </p>
+        </div>
+      </div>
+    </section>
+    <section class="section-white">
+      <div class="container">
+        <div class="column__full">
+          <h2>Exemple de requête</h2>
+          <p>
+            L'adresse à taper pour une requête est : <strong>Adresse + Version + Informations</strong><br>
```
Comment (thimy) [inline src/components/ApiDoc.vue]: "interroger" plutôt que "taper" ?

File: src/components/ApiDoc.vue [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+            Exemple : Je souhaite accéder aux informations sur la mairie de Montpellier, dont le siret est le 21340172201787. Je me rend sur <a href="https://sirene.entreprise.api.gouv.fr/v1/siret/21340172201787/">https://sirene.entreprise.api.gouv.fr/v1/siret/21340172201787</a>
```
Comment (thimy) [inline src/components/ApiDoc.vue]: rends*

File: src/components/ApiDoc.vue [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+          <a class="button" href="https://www.apientreprise.fr/">Demander un accès</a>
+        </div>
+      </div>
+    </section>
+  </div>
+</template>
+
+<script>
+import ApiDocInput from '@/components/apiDoc/ApiDocInput'
+// import ApiDocOutput from '@/components/apiDoc/ApiDocOutput'
```
Comment (thimy) [inline src/components/ApiDoc.vue]: Remove comments

File: src/components/ApiDoc.vue [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+    this.$store.dispatch('hideWelcomeText')
+  }
+}
+
+</script>
+
+<style lang="scss" scoped>
+
+section {
+  padding: 20px, 0, 20px, 0;
```
Comment (thimy) [inline src/components/ApiDoc.vue]: No need for commas

File: src/components/ApiDoc.vue [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+}
+
+p {
+  min-width: 100%
```
Comment (thimy) [inline src/components/ApiDoc.vue]: Not sure about this min-width 🤔

File: src/components/ApiDoc.vue [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+  margin: 1em 0 1em 0;
+}
+
+h3 {
+  margin-bottom: 0;
+}
+
+// Table style
+table {
+  margin: 25px auto;
```
Comment (thimy) [inline src/components/ApiDoc.vue]: You should set a white background to this table.

File: src/components/Search.vue [hunk 7]
```diff
@@ -1,50 +1,38 @@
...
+          <h1 class="search__title">
+            Retrouvez toutes les informations concernant les entreprises de France
+          </h1>
+          <p class="search__subtitle">La plus grande base de données sur l'état civil des entreprises françaises est maintenant accessible à tous, sans frais.</p>
+        </div>
+      </transition>
+      <SearchBar searchName="Recherche par nom"></SearchBar>
+      <router-link v-if="showBackToResultsButton" class="back-to-results" :to="{ path: '/search',
+        query: {
+          fullText: this.$store.state.search.storedFullText,
```
Comment (thimy) [inline src/components/Search.vue]: You should put these params in data and use getters.

File: src/components/home/PublicAdministration.vue [hunk 8]
```diff
@@ -3,7 +3,7 @@
     <div class="container">
         <p>
           Vous êtes une administration publique ?<br />
-          Demandez votre accès spécial à l'API pour obtenir des informations supplémentaires sur les sociétés.
+          Demandez votre accès spécial à API Entreprise pour obtenir des informations supplémentaires sur les sociétés
         </p>
         <a class="button" href="https://www.apientreprise.fr/">Demander un accès</a>
```
Comment (thimy) [inline src/components/home/PublicAdministration.vue]: I think you can change for entreprise.api.gouv.fr ;)

File: src/components/ApiDoc.vue [hunk 9]
```diff
@@ -0,0 +1,369 @@
...
+          <h2>Comment fonctionne Entreprise.data.gouv.fr ?</h2>
+          <p>
+            Ce site web interroge l'API Sirene développée par <a href="https://www.data.gouv.fr/fr/organizations/etalab/">Étalab</a>.
+            Ce site web et son API sont développés en open-source : vous pouvez <a href="https://github.com/betagouv/sirene_as_api">télécharger le code sur GitHub</a>.
+          </p>
+          <h2>Puis-je interroger directement l'API Sirene ?</h2>
+          <p>Bien sûr ! Pas besoin d'être développeur. Cette documentation est faite pour vous.</p>
+        </div>
+        <div class="column__icon">
+          <img src="../assets/img/icons/three-gears.svg" alt="gears">
```
[the remark this answers is not in the document]
Comment (Samuelfaure) [reply src/components/ApiDoc.vue]: I  went and read the guidelines asking that alt be empty for illustrations with no informations

**Pull request merged**